freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/user/event/SetPasswordRequest.java

```diff
@@ -9,14 +9,17 @@
 
     private final String username;
 
+    private final String userCrn;
+
     private final String password;
 
     private final Optional<Instant> expirationInstant;
 
-    public SetPasswordRequest(Long stackId, String environment, String username, String password, Optional<Instant> expirationInstant) {
+    public SetPasswordRequest(Long stackId, String environment, String username, String userCrn, String password, Optional<Instant> expirationInstant) {
```

Comment:
I assume that the userCrn is the actor for which the password is set? and that is extracted from the authentication context? Maybe call this actorCrn as the caller maybe a machine user? Also, how do you know the username at this point? This is the workload username, right?

Reply:
Yes userCrn is the actor for which the password is set and is pulled from authentication context. Changing it to actorCrn does make sense though I'm seeing userCrn be used for both user and machine user in other places in this package. I think we can leave at as userCrn for now, since we will probably deprecate this path once single user sync is in.

The username is the workload username, the existing code calls ums to get the workload username from ums. See PasswordService.getUserIdFromUserCrn(...)